server/plugins/LFUGens.cpp

```diff
@@ -402,23 +399,30 @@ void Vibrato_next(Vibrato* unit, int inNumSamples) {
 
 void Vibrato_Ctor(Vibrato* unit) {
     unit->mFreqMul = 4.0 * SAMPLEDUR;
-    unit->mPhase = 4.0 * sc_wrap(ZIN0(7), 0.f, 1.f) - 1.0;
+    double initPhase = unit->mPhase = 4.0 * sc_wrap(ZIN0(7), 0.f, 1.f) - 1.0;
```

Comment:
Since these aren't mutated, its good practice to declare these variables `const double` https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rconst-immutable

This happens throughout.

I mean, I've never seen actual code guidelines for supercollider, so I guess you could ignore this?

Reply:
I agree in principle, though I think the prevailing approach in the cpp codebase, when making small local changes, is to stick with the surrounding convention, making changes that are only pertinent to the functional changes (e.g. not adding `const` to all the variables in this local function).

Reply:
Sure, but in this case, it's a new variable that wasn't there before. Adding const would only change your code.

I don't think following the style of the surrounding code is good practice when it's harder to read, uses unsafe or discouraged features. Following the surrounding code over the recommendations of the language's creator seems problematic when applied in general.

Reply:
I don't disagree, again, it's really a matter of convention. Is this code meaningfully more safe?
```
void Vibrato_Ctor(Vibrato* unit) {
    unit->mFreqMul = 4.0 * SAMPLEDUR;
    double initPhase = unit->mPhase = 4.0 * sc_wrap(ZIN0(7), 0.f, 1.f) - 1.0;

    RGen& rgen = *unit->mParent->mRGen;
    float rate = ZIN0(1) * unit->mFreqMul;
    float depth = ZIN0(2);
    float rateVariation = ZIN0(5);
    float depthVariation = ZIN0(6);
    float initFreq = unit->mFreq = rate * (1.f + rateVariation * rgen.frand2());
    float initScaleA = unit->m_scaleA = depth * (1.f + depthVariation * rgen.frand2());
    float initScaleB = unit->m_scaleB = depth * (1.f + depthVariation * rgen.frand2());
    const float someNewVar = 3.f;

    unit->m_delay = (int)(ZIN0(3) * SAMPLERATE);
    unit->m_attack = (int)(ZIN0(4) * SAMPLERATE);
    unit->m_attackSlope = 1. / (double)(1 + unit->m_attack);
    unit->m_attackLevel = unit->m_attackSlope;
    unit->trig = 0.0f;
    SETCALC(Vibrato_next);
    Vibrato_next(unit, 1);

    unit->mPhase = initPhase;
    unit->mFreq = initFreq;
    unit->m_scaleA = initScaleA;
    unit->m_scaleB = initScaleB;
    unit->trig = 0.0f;
}
```

Reply:
Yea I am, as I mentioned at the beginning it's only really the c cast that must be changed. It's fine as is.

Reply:
I've pushed the updated version with the const declaration, maybe we start a trend going forward :)